File: Arrays/MergeIntervals/MergeIntervals.java
package Arrays.MergeIntervals;
import java.util.*;
public class MergeIntervals {

    public static int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, new Comparator<int[]>(){
            public int compare(int[] it, int[] it2){
                return it[0]-it2[0];
            }
        });
        List<int[]> list = new ArrayList<>();
        int st=intervals[0][0], end=intervals[0][1];
        int i,j,n=intervals.length;
        for(i=1;i<n;i++){
            if(intervals[i][0]<=end){
                end = Math.max(end, intervals[i][1]);
            }
            else{
                int[] temp = new int[2];
                temp[0] = st;
                temp[1] = end;
                list.add(temp);
                st = intervals[i][0];
                end = intervals[i][1];
            }
        }
        int[] temp = new int[2];
                temp[0] = st;
                temp[1] = end;
                list.add(temp);
        j=0;
        int[][] res = new int[list.size()][];
        for(int[] it: list){
            res[j++] = it;
        }
        System.out.println("list is "+ list);
        return res;
    }
    public static void main(String[] args) {
        
    }
}

File: Arrays/MergeIntervals/MergeIntervals.go
package main

import (
	"fmt"
	"sort"
)

func mergeIntervals(arr [][]int) [][]int {
	sort.Slice(arr, func(i, j int) bool {
		return arr[i][0] < arr[j][0]
	})
	fmt.Println(arr)
	intervals := make([][]int, 0)
	n := len(arr)
	start := arr[0][0]
	end := arr[0][1]
	for i := 1; i < n; i++ {
		if arr[i][0] <= end {
			if arr[i][1] > end {
				end = arr[i][1]
			}
		} else {
			intervals = append(intervals, []int{start, end})
			start = arr[i][0]
			end = arr[i][1]
		}
	}
	intervals = append(intervals, []int{start, end})
	return intervals
}

func main() {
	arr := [][]int{{1, 4}, {0, 4}}
	fmt.Println(mergeIntervals(arr))
}
